frontend/src/pages/PublicCardsPage.tsx

```diff
@@ -34,18 +31,46 @@ const PublicCardsPage = () => {
     cards,
     cardCount,
     tags,
+    heartInfo,
+    setHeartInfo,
     isAllCardChecked,
     checkAllCard,
     checkedCardCount,
     checkCard,
     takeCardsToMyWorkbook,
     isLoading,
+    toggleHeart,
   } = usePublicCard();
 
-  const showSnackbar = useSnackbar();
   const { openModal, closeModal } = useModal();
   const { routeQuiz } = useRouter();
 
+  const { heart, heartCount, serverHeart } = heartInfo;
+
+  const onClickHeart = () => {
+    debounce(() => {
+      if (serverHeart === !heart) return;
+
+      toggleHeart();
+    }, 200);
+
+    if (heart) {
+      setHeartInfo((prevValue) => ({
+        ...prevValue,
+        heart: false,
+        heartCount: heartCount - 1,
+      }));
+
+      return;
+    }
+
+    setHeartInfo((prevValue) => ({
+      ...prevValue,
+      heart: true,
+      heartCount: heartCount + 1,
+    }));
+  };
```

Comment:
```suggestion
setHeartInfo((prevValue) => ({
      ...prevValue,
      heart: !heart,
      heartCount: heart ? heartCount - 1 : heartCount + 1,
    }));
```
은 어떨까요? 흐름을 제대로 파악하지 못해 이렇게 썼지만, 함수형으로 상태를 set해준다면, `prevValue.heart` 나 `prevValue.heartCount`도 참조해 볼 수 있지 않을까 싶네요 ㅎㅎ

Reply:
오오오.. 천재 변경할게요~